tests/e2e/e2e_test.go

```diff
@@ -368,3 +384,340 @@ func (s *IntegrationTestSuite) verifyChainPassesUpgradeHeight(c *chain, valIdx i
 		5*time.Second,
 	)
 }
+
+
+
+// globalfee in genesis is set to be "0.00001photon"
+func (s *IntegrationTestSuite) TestQueryGlobalFeesInGenesis() {
+	chainAAPIEndpoint := fmt.Sprintf("http://%s", s.valResources[s.chainA.id][0].GetHostPort("1317/tcp"))
+	feeInGenesis, err := sdk.ParseDecCoins(initialGlobalFeeAmt+photonDenom)
+	s.Require().NoError(err)
+	s.Require().Eventually(
+		func() bool {
+			fees, err := queryGlobalFees(chainAAPIEndpoint)
+			s.T().Logf("Global Fees in Genesis: %s", fees.String())
+			s.Require().NoError(err)
+
+		return fees.IsEqual(feeInGenesis)
+		},
+		15*time.Second,
+		5*time.Second,
+	)
+}
+
+/*
+global fee in genesis is "0.00001photon", which is the same as min_gas_price.
+This initial value setup is for not to fail other e2e tests.
+global fee e2e tests:
+0. initial globalfee = 0.00001photon, min_gas_price = 0.00001photon
+
+test1. gov proposal globalfee = [], query globalfee still get empty
+   - tx with fee denom photon, fail
+   - tx with zero fee denom photon, pass
+   - tx with fee denom uatom, pass
+   - tx with fee empty, pass
+
+test2. gov propose globalfee =  0.000001photon(lower than min_gas_price)
+   - tx with fee higher than 0.000001photon but lower than 0.00001photon, fail
+   - tx with fee higher than/equal to 0.00001photon, pass
+   - tx with fee uatom fail
+
+test3. gov propose globalfee = 0.0001photon (higher than min_gas_price)
+   - tx with fee equal to 0.0001photon, pass
+   - tx with fee equal to 0.00001photon, fail
+
+test4. gov propose globalfee =  0.000001photon (lower than min_gas_price), 0uatom
+   - tx with fee 0.000001photon, fail
+   - tx with fee 0.000001photon, pass
+   - tx with empty fee, pass
+   - tx with fee uatom pass
+   - tx with fee 0uatom, 0.000005photon fail
+   - tx with fee 0uatom, 0.00001photon pass
+5. check balance correct: all the sucessful tx sent token amt is received
+6. gov propose change back to initial globalfee = 0.00001photon, This is for not influence other e2e tests.
+*/
+func (s *IntegrationTestSuite) TestGlobalFees() {
```

Comment:
We should probably start moving these into their own files

Reply:
what does this mean ?

Reply:
let's add it to the refactor issue.
@yaruwangway i think she means that we could have a separate file for each module to keep this file from being too large.